src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -709,6 +910,13 @@ namespace AppInstaller::Repository
         AICLI_LOG(Repo, Info, << "Source created with extra data: " << details.Data);
         AICLI_LOG(Repo, Info, << "Source created with identifier: " << details.Identifier);
 
+        if (!IsAddedUserSourceAllowedByPolicy(details))

```

Comment:
>if (!IsAddedUserSourceAllowedByPolicy(details)) [](start = 8, length = 47)

This check cannot be done before AddSourceFromDetails?  Type + Arg should be enough to block then since it is an allow list.

Reply:
The policy includes the source data and identifier. (Because we need it for additional sources and it is easier to be consistent.)
Since we have the data and identifier, I think we might as well use it and verify that it actually matches. And that can only be after the factory extracts them.

I don't expect this check to fail often because I'm actually checking the Type + Arg before.

I'm making a lot of assumptions here (that we should require having the Data and Identifier, that it is possible they don't match even if the Type + Arg do, that we actually care if they don't, that we cannot verify it without adding the source). I'm happy to remove it if you don't think any of those is true.

Reply:
If the Type+Arg check is happening before we actually throw off to the source code then it should be fine.

Reply:
Uh... Do you mean "it should be fine" as in "the source should be fine and we don't need to check here" or as in "it should be fine to check here as long as we also do it before"?

Reply:
We don't need a check in this function as long as one is happening in the stack before ISourceFactory::Add.